tests/pytests/integration/utils/test_smb.py

```diff
@@ -0,0 +1,378 @@
+    ),
+    pytest.mark.skip_if_binaries_missing("smbd", check_all=False),
+    pytest.mark.skip_unless_on_linux(reason="using Linux samba to test smb"),
+]
+
+
+def check_pid(pid):
+    try:
+        os.kill(pid, 0)
+    except OSError:
+        return False
+    else:
+        return True
+
+
+@pytest.fixture
+def smb_dict(tmp_path):
+    samba_dir = tmp_path / "samba"
+    samba_dir.mkdir(parents=True)
+    assert samba_dir.exists()
+    assert samba_dir.is_dir()
+    public_dir = tmp_path / "public"
+    public_dir.mkdir(parents=True)
+    assert public_dir.exists()
+    assert public_dir.is_dir()
+    passwdb = tmp_path / "passwdb"
+    username = getpass.getuser()
+    with salt.utils.files.fopen(passwdb, "w") as fp:
+        fp.write(
+            f"{username}:0:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX:AC8E657F8"
+            "3DF82BEEA5D43BDAF7800CC:[U          ]:LCT-507C14C7:"
+        )
+    samba_conf = tmp_path / "smb.conf"
+    with salt.utils.files.fopen(samba_conf, "w") as fp:
+        fp.write(
+            f"[global]\n"
+            "realm = saltstack.com\n"
+            "interfaces = lo 127.0.0.0/8\n"
+            "smb ports = 1445\n"
+            "log level = 2\n"
+            "map to guest = Bad User\n"
+            "enable core files = no\n"
+            "passdb backend = smbpasswd\n"
+            f"smb passwd file = {passwdb}\n"
+            f"lock directory = {samba_dir}\n"
+            f"state directory = {samba_dir}\n"
+            f"cache directory = {samba_dir}\n"
+            f"pid directory = {samba_dir}\n"
+            f"private dir = {samba_dir}\n"
+            f"ncalrpc dir = {samba_dir}\n"
+            "socket options = IPTOS_LOWDELAY TCP_NODELAY\n"
+            "min receivefile size = 0\n"
+            "write cache size = 0\n"
+            "client ntlmv2 auth = no\n"
+            "client min protocol = SMB3_11\n"
+            "client plaintext auth = no\n"
+            "\n"
+            "[public]\n"
+            f"path = {public_dir}\n"
+            "read only = no\n"
+            "guest ok = no\n"
+            "writeable = yes\n"
+            f"force user = {username}\n"
+        )
+
+    ## _smbd = subprocess.Popen([shutil.which("smbd"), "-F", "-P0", "-s", samba_conf])
+    ## _smbd = subprocess.Popen([smbd_path, "-F", "-P0", "-s", samba_conf])
+    ## time.sleep(2)
+    ## pidfile = samba_dir / "smbd.pid"
+    ## conn_dict = {
+    ##     "tmpdir": tmp_path,
+    ##     "samba_dir": samba_dir,
+    ##     "public_dir": public_dir,
+    ##     "passwdb": passwdb,
+    ##     "username": username,
+    ##     "samba_conf": samba_conf,
+    ##     "smbd_path": smbd_path,
+    ##     "pidfile": pidfile,
+    ## }
+
+    ## assert pidfile.exists()
+
+    smbd_path = shutil.which("smbd")
+    pathlib.Path(smbd_path).exists()
+    try:
+        _smbd = subprocess.Popen(
+            [smbd_path, "-i", "-d", "2", "-F", "-P0", "-s", samba_conf]
+        )
+        assert _smbd != 0
```

Comment:
```suggestion
        assert _smbd != 0
```
This is always true because `_smdb` is a [`Popen`](https://docs.python.org/3.5/library/subprocess.html#subprocess.Popen) instance.

Reply:
code changed